### ISSUE TITLE
incorrect rewriting of link to `/`

### ISSUE THREAD

**Issue by rgaudin** [openzim/zimwriterfs#139]
When linking to `/`, zimwriterfs rewrites it to `/I`.

Example:

``` html
<a href="/">hello</a>
becomes
<a href="/I">hello</a>
```

**Note:** linking to `/` from within a ZIM is a bad practice as the result is unpredicatble and probably not wanted but I see no reason for it to rewrite to `/I`.

**reproduce with:**

``` sh
mkdir -p build
wget -O build/favicon.png http://download.kiwix.org/other/youscribe/favicon.png
printf '<!DOCTYPE html><html>\n<head><meta charset="utf-8" /><title>test</title><link id="favicon" rel="icon" href="favicon.png" type="image/png" /></head>\n<body><a href="/"><img src="favicon.png" /></a><p>nothing</p></body>\n</html>\n' > build/home.html
zimwriterfs -w home.html -f favicon.png -l eng -t test -d test -c kiwix -p kiwix build/ link-err.zim
```

**Comment by kelson42** [comment]
@rgaudin I'm not sure this is valid bug. `zimwriterfs` packs a fully working static web site stored on the filesystem. This implies that all links in the HTML are relative. If you put a `/`, then this is absolute and means the the local web site is "broken".

**Comment by rgaudin** [comment]
Agrees but then it should not modify it right? Don't think it's important but at least now it's documented.

**Comment by kelson42** [comment]
@rgaudin It should definitly print a warning or similar... and yes, it should probably be kept.

**Comment by jayanthj737** [comment]
Is this an issue?

**Comment by kelson42** [comment]
Yes, the `/` should be kept and a warning should be printed.

**Comment by asashnov** [comment]
If link ```<a href="/">hello</a>``` on a regular website (like http://example.com/) will lead to homepage (the root of site), then it's logically to point to welcome page (specified with -w)

**Comment by kelson42** [comment]
@asashnov What happen in general if we have an absolute link (so any link starting with "/")?

**Comment by asashnov** [comment]
> @asashnov What happens in general if we have an absolute link (so any link starting with "/")?

Absolute web URLs remains unchanged: https://example.com/ remains the same.

Absolute local paths:

Links to directories go to /I namespace:

/  ->  /I
/dir  ->  /I/dir

HTMLs are in global namespace (should it be in /A/ ?):

/hello.html  -> /hello.html
/dir/hello.html ->  /dir/hello.html

JPG and PNG are put to /I (this is correct):

/1.png  -> /I/1.png
/dir/1.png  ->  /I/dir/1.png

[home.html.zip](https://github.com/openzim/zimwriterfs/files/4708001/home.html.zip)

**Comment by asashnov** [comment]
Also I checked if I setup example.com website on my machine and put 

test.html:
```html
<html>
<body>
<p><a href="/">to site root</a>
</p>
</body>
</html>
```

The link <a href="/"  leads to the main page: http://example.com/ in browser.

So it would replicate website and web browser behaviour if we change such links to the welcome page.

(the instruction https://www.digitalocean.com/community/tutorials/how-to-set-up-nginx-virtual-hosts-server-blocks-on-ubuntu-12-04-lts--3 is still actual for Ubuntu 18.04)

**Comment by kelson42** [comment]
@asashnov https://example.com/ is not what I mean with an absolute URL. I mean anything starting with `/`.

**Comment by kelson42** [comment]
@asashnov Ok, so please for *any absolute URLs* (so any link src/href starting with `/`):
* Print a warning on the console
* Just leave them like they are (so no URL rewritting).

**Comment by mgautierfr** [comment]
> Just leave them like they are (so no URL rewritting)

I disagree with that. The spec explicitly says that `Absolute URLs, ie. with a leading slash (/) are forbidden, because this avoid including the ZIM contents in any HTTP sub-hierachy. ZIM contents URLs must consequently be relative.`(https://www.openzim.org/wiki/ZIM_file_format)

I must not allow our tool to write an invalid zim.

The url must be changed to be relative and the (corresponding) relative url must resolved to the something equivalent to the absolute url.

**Comment by kelson42** [comment]
Ok, so lets print an error and stop then. Migh bee even better indeed.
